Nodejs/Product/Nodejs/Debugger/NodeDebugger.cs

```diff
@@ -243,8 +243,16 @@ public NodeDebugger(Uri debuggerEndpointUri, int id)
             if (asyncBreakComplete != null) {
                 asyncBreakComplete(this, new ThreadEventArgs(MainThread));
             }
-        }
-
+        }
+
+        internal bool IsRunning() {
+            var backtraceCommand = new BacktraceCommand(CommandId, _resultFactory, 0, 1);
+            if (TrySendRequestAsync(backtraceCommand).GetAwaiter().GetResult()) {

```

Comment:
I see some of the below commands have a timeout set. Is there potential this will block? For how long? Is there a chance this will be on the UI thread?